### PR TITLE
Add account balances

### PR DESCRIPTION
**Pull request by duyentb95**
Method
1) FEE SUSd - Sum SUSd balanceOf for address: 0xfeEFEEfeefEeFeefEEFEEfEeFeefEEFeeFEEFEeF read from optimism + ethereum
2) User SUSd -Sum SUSd balanceOf for address: user address read from optimism
3) User SDS -Sum SDS balanceOf for address: user address read from optimism + ethereum
4) Total SDS- Sum SDS totalSupply on optimism + eth

Total Fees:
FEE SUSd = Total fees collected this epoch

Equation:
Fee SUSd * (User SDS) / (Total SDS) = Percentage of fee's to user address
======================================
Method
1) User SDS -Sum SDS balanceOf for address: user address read from optimism + ethereum
2) Total SDS- Sum SDS totalSupply on optimism + eth
3) User SNX - Sum SNX balanceOf for address: user address read from optimism + ethereum
4) User Escrow SNX - Sum Escrow SNX balanceOf for address: user address read from optimism + ethereum
4) protocol issuanceRatio -  (issuanceRatio / 1e18)
5) SDS Price - latestAnswer / 1e27
6) SNX Price - latestAnswer / 1e8
equation: 
user issuanceRatio = (SNX Price * (User SNX + User Escrow SNX) / (User SDS * SDS Price)))

Check if user issuanceRatio > .2, if it's greater then default it to .2. else set it equal to calculated user issuanceRatio

Equation:
(protocol issuanceRatio / user issanceRatio) * (User SNX + User Escrow SNX) = amount snx staked

### CODE REVIEW

File: ui/src/hooks/contracts/sds.ts [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+    name: 'NewTransmission',
+    type: 'event',
+  },
+  {
+    anonymous: false,
+    inputs: [
+      { indexed: true, internalType: 'address', name: 'transmitter', type: 'address' },
+      { indexed: true, internalType: 'address', name: 'payee', type: 'address' },
+      { indexed: false, internalType: 'uint256', name: 'amount', type: 'uint256' },
+      {
```
Comment (jmzwar) [inline ui/src/hooks/contracts/sds.ts:140]: Do we need this abi file if we import directly from the contracts?

File: ui/src/components/WalletBalances/WalletBalances.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      {/*  label={'SNX Staked: '} */}
+      {/*  value={snxStakedAmount} */}
+      {/*  tooltip={ */}
+      {/*    <Flex alignItems="flex-start" flexDirection="column"> */}
+      {/*      <Text>On Optimism + Ethereum</Text> */}
+      {/*      <Text>Total Debt: ${formatNumber(totalDebt)}</Text> */}
+      {/*    </Flex> */}
+      {/*  } */}
+      {/* /> */}
+      <Row label={'Current Estimated Epoch Fees: '} value={estimatedUserFee} prefix={'$'} />
```
Comment (jmzwar) [inline ui/src/components/WalletBalances/WalletBalances.tsx:78]: I think a tooltip here is in order - just to specify that the fees are coming from SNX staking and not from trading fees. "Fees earned from SNX Staking"

File: ui/src/hooks/contracts/sds.ts (hunk 1 above)
Comment (jmzwar) [reply ui/src/hooks/contracts/sds.ts:140]: Ah pls ignore this looks like chainlink

**Review by jmzwar — CHANGES_REQUESTED** [review]
Cool so I think let's add a tooltip and LGTM

File: ui/src/components/WalletBalances/WalletBalances.tsx (hunk 2 above)
Comment (duyentb95) [reply ui/src/components/WalletBalances/WalletBalances.tsx]: I also think it's a good idea and I have added it.

File: ui/src/hooks/contracts/sds.ts (hunk 1 above)
Comment (duyentb95) [reply ui/src/hooks/contracts/sds.ts:140]:
```
{
    inputs: [],
    name: 'latestAnswer',
    outputs: [{ internalType: 'int256', name: '', type: 'int256' }],
    stateMutability: 'view',
    type: 'function',
  },
```
```
  const snxContract = new Contract(snxPoolAddressMainnet, snxAbi, globalProviders.mainnet);
      const sdsContract = new Contract(sdsPoolAddressMainnet, sdsAbi, globalProviders.mainnet);

      const [snxPriceBn, sdsPriceBn] = await Promise.all([
        snxContract.latestAnswer(),
        sdsContract.latestAnswer(),
      ]);
```
  Need to call `latestAnswer`

**Pull request merged**